src/datasource.ts

```diff
@@ -296,22 +303,34 @@ export default class CogniteDatasource extends DataSourceApi<
     );
   }
 
-  async fetchEvents(expr: string, timeRange: EventsFilterTimeParams) {
-    const { filters, params } = parseQuery(expr);
+  async fetchEvents({ expr, eventQuery }, timeRange: EventsFilterTimeParams) {
+    let filter = [];
+    let params = {};
+    if (expr) {
+      const parsedQuery = parseQuery(expr);
+      filter = parsedQuery.filters;
+      params = parsedQuery.params;
+    }
+    const advancedFilter =
+      this.connector.isEventsAdvancedFilteringEnabled() && eventQuery.length
```

Comment:
```suggestion
      this.connector.isEventsAdvancedFilteringEnabled() && eventQuery
```
this is safer, and cleaner